x-pack/test/functional/apps/ml/data_visualizer/index_data_visualizer.ts

```diff
@@ -326,105 +345,60 @@ export default function ({ getService }: FtrProviderContext) {
-          testData.metricFieldsFilter,
-          testData.expected.metricFieldsFilterCardCount
+      await ml.testExecution.logTestStep(
+        'displays details for the created job in the analytics table'
+      );
+      for (const fieldCard of testData.expected.metricCards as Array<
+        Required<MetricFieldVisConfig>
+      >) {
+        await ml.dataVisualizerTable.assertNumberRowContents(
+          fieldCard.fieldName,
+          fieldCard.docCountFormatted
         );
       }
 
-      if (
-        testData.expected.nonMetricCards !== undefined &&
-        testData.expected.nonMetricCards.length > 0
-      ) {
-        await ml.testExecution.logTestStep(
-          `${testData.suiteTitle} displays the non-metric Fields panel`
-        );
-        await ml.dataVisualizerIndexBased.assertFieldsPanelForTypesExist(
-          getFieldTypes(testData.expected.nonMetricCards as FieldVisConfig[])
-        );
-
-        await ml.testExecution.logTestStep(
-          `${testData.suiteTitle} displays the expected non-metric field cards and contents`
-        );
-        for (const fieldCard of testData.expected.nonMetricCards!) {
-          await ml.dataVisualizerIndexBased.assertCardExists(fieldCard.type, fieldCard.fieldName);
-          await ml.dataVisualizerIndexBased.assertNonMetricCardContents(
-            fieldCard.type,
-            fieldCard.fieldName!,
-            fieldCard.exampleCount
-          );
-        }
-
-        await ml.testExecution.logTestStep(
-          `${testData.suiteTitle} sets the non metric field types input`
-        );
-        const fieldTypes: ML_JOB_FIELD_TYPES[] = getFieldTypes(
-          testData.expected.nonMetricCards as FieldVisConfig[]
-        );
-        await ml.dataVisualizerIndexBased.assertFieldsPanelTypeInputExists(fieldTypes);
-        await ml.dataVisualizerIndexBased.setFieldsPanelTypeInputValue(
-          fieldTypes,
-          testData.nonMetricFieldsTypeFilter,
-          testData.expected.nonMetricFieldsTypeFilterCardCount
-        );
-
-        await ml.testExecution.logTestStep(
-          `${testData.suiteTitle} filters non-metric fields cards with search`
-        );
-        await ml.dataVisualizerIndexBased.filterFieldsPanelWithSearchString(
-          fieldTypes,
-          testData.nonMetricFieldsFilter,
-          testData.expected.nonMetricFieldsFilterCardCount
-        );
+      for (const fieldCard of testData.expected.nonMetricCards as Array<
+        Required<MetricFieldVisConfig>
+      >) {
+        await ml.dataVisualizerTable.assertRowExists(fieldCard.fieldName);
+      }
 
-        await ml.testExecution.logTestStep(
-          `${testData.suiteTitle} sample size control changes non-metric field cards doc count`
-        );
-        await ml.dataVisualizerIndexBased.clearFieldsPanelSearchInput(fieldTypes);
-        await ml.dataVisualizerIndexBased.assertSampleSizeInputExists();
-        await ml.dataVisualizerIndexBased.setSampleSizeInputValue(
```

Comment:
Taking this line as an example for a few things.
I'm a bit worried about losing test coverage here as the following user actions are not tested anymore:
- Filter table with the search bar
- Filter table with the field type input
- Change the sample size

I think it would be good to merge without losing test coverage, but not sure if this is in the scope of this PR or if this should be done in a follow-up. @peteharverson could you please assess?

Also, there are a couple `ml.dataVisualizerIndexBased` service methods that are not needed anymore (or need to be adjusted to cover the user actions mentioned above):
- assertFieldsPanelForTypesExist
- assertCardExists
- assertCardContentsExists
- assertNonMetricCardContents
- assertDocumentCountCardContents
- assertNumberCardContents
- assertDateCardContents
- assertKeywordCardContents
- assertTextCardContents
- assertFieldDocCountExists
- assertFieldDocCountContents
- assertFieldCardinalityExists
- assertNumberStatsContents
- assertTopValuesContents
- assertFieldsPanelCardCount
- assertFieldsPanelSearchInputValue
- clearFieldsPanelSearchInput
- filterFieldsPanelWithSearchString
- assertFieldsPanelTypeInputExists
- assertFieldsPanelTypeInputValue
- setFieldsPanelTypeInputValue
- setSampleSizeInputValue

Reply:
We definitely don't want to lose test coverage, but I'm happy to merge this PR without those three tests @pheyos highlighted, and to add them back in a subsequent PR.